servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/extensions/collectionExtensions.kt

```diff
@@ -0,0 +1,9 @@
+package com.expediagroup.graphql.server.extensions
+
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.supervisorScope
+
+suspend fun <A, B> Iterable<A>.concurrentMap(transform: suspend (A) -> B): List<B> = supervisorScope {
+    map { async { transform(it) } }.awaitAll()
```

Comment:
if any of the requests fail (they shouldn't as underlying failures should be handled), we should create generic response with error